core/src/main/java/com/linecorp/armeria/common/util/BlockingTaskExecutorBuilder.java

```diff
@@ -0,0 +1,140 @@
+    public BlockingTaskExecutorBuilder numThreads(int numThreads) {
+        checkArgument(numThreads >= 0, "numThreads: %s (expected: >= 0)", numThreads);
+        this.numThreads = numThreads;
+        return this;
+    }
+
+    /**
+     * Sets the amount of keep alive time in seconds.
+     */
+    public BlockingTaskExecutorBuilder keepAliveTime(Duration keepAliveTime) {
+        checkArgument(!requireNonNull(keepAliveTime, "keepAliveTime").isNegative(),
+                      "keepAliveTime: %s (expected: >= 0)");
+        return keepAliveTimeMillis(keepAliveTime.toMillis());
+    }
+
+    /**
+     * Sets the amount of keep alive time in seconds.
+     */
+    public BlockingTaskExecutorBuilder keepAliveTimeMillis(long keepAliveTimeMillis) {
+        checkArgument(keepAliveTimeMillis >= 0, "keepAliveTimeMillis: %s (expected: >= 0)", keepAliveTimeMillis);
+        this.keepAliveTimeMillis = keepAliveTimeMillis;
+        return this;
+    }
+
+    /**
+     * Sets the flag of daemon for new threads.
+     */
+    public BlockingTaskExecutorBuilder daemon(boolean daemon) {
+        this.daemon = daemon;
+        return this;
+    }
+
+    /**
+     * Sets the priority for new threads.
+     */
+    public BlockingTaskExecutorBuilder priority(int priority) {
+        checkArgument(priority >= Thread.MIN_PRIORITY && priority <= Thread.MAX_PRIORITY,
+                      "priority: %s (expected: Thread.MIN_PRIORITY <= priority <= Thread.MAX_PRIORITY)",
+                      priority);
+
+        this.priority = priority;
+        return this;
+    }
+
+    /**
+     * Use this method to set additional work before or after the Runnable is run. For example:
+     * <pre>{@code
+     * ThreadFactories.builder("thread-prefix")
+     *                .taskFunction( task -> {
+     *                    return () -> {
+     *                        // Add something to do before task is run
+     *                        task.run();
+     *                        // Add something to do after task is run
+     *                    };
+     *                })
+     *                .build();
+     * }</pre>
+     *
+     * Sets the task function for new threads.
+     */
+    public BlockingTaskExecutorBuilder taskFunction(
+            Function<? super Runnable, ? extends Runnable> taskFunction) {
+        this.taskFunction = requireNonNull(taskFunction, "taskFunction");
+        return this;
+    }
+
+    /**
+     * Returns a newly-created {@link BlockingTaskExecutor} with the properties given so far.
+     */
+    public BlockingTaskExecutor build() {
+        final ThreadFactory threadFactory = ThreadFactories.builder(threadNamePrefix)
+                                                           .daemon(daemon)
+                                                           .priority(priority)
+                                                           .taskFunction(taskFunction)
+                                                           .build();
```

Comment:
This builder duplicates a lot with `ThreadFactoryBuilder`. Do we need an abstract layer for them? 
```java
abstract class AbstractThreadFactoryBuilder {
    // Add common builders here
    
   ThreadFactory buildFactory() {
       ...
   }
}

public class ThreadFactoryBuilder extends AbstractThreadFactoryBuilder {
   
    ...
    public ThreadFactoryBuilder eventLoop(boolean eventLoop) {
    }
}

public class BlockingTaskExecutorBuilder extends AbstractThreadFactoryBuilder {
   public BlockingTaskExecutor build() {
       final ThreadFactory threadFactory = buildThreadFactory();
       ...
   }
}
```

Reply:
I understand that this `BlockingTaskExecutorBuilder` will be use for generating `BlockingTaskExecutor` purpose only and has a plan to extend the `BlockingTaskExecutor` interface to fit armeria's functionalities, am I correct? 

To add such feature, I am not sure utilizing suggested builder is the right way.

Also, if we decide to move this way, we need to decide not-intersected methods factory. (allowing event loop thread creation)

Reply:
On second thought, the dupes are acceptable. Let's keep it as it is. We can easily refactor this later when we really need it.